<?xml version="1.0"?>
<launch>
  <include file="$(find gazebo_ros)launch/empty_world.launch">
      <arg name="world_name" value="$(find example_package)/world/playground.world" />
      <arg name="gui" value="true" />
      <arg name="use_sim_time" value="true" />
      <arg name="paused" value="false" />
  </include>

  <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_burger_for_autorace.urdf.xacro" /> 
  <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf" args="-urdf -model turtlebot3_burger -x 0 -y 0 -z 0 -Y 0 -param robot_description" />
  <include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch"/>


  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find example_package)/config/config.rviz" />

  <node pkg="turtlebot3_teleop" type="turtlebot3_teleop_key" name="turtlebot3_teleop_key" launch-prefix="xterm -e"/>


  <!-- KF -->
  <node name="kf_node" pkg="example_package" type="filter_node_KF" output="screen" />

  <!-- EKF -->
  <node name="ekf_node" pkg="example_package" type="filter_node_EKF" output="screen">
    <param name="scan_noise_std" value="0.05" />
    <param name="imu_yaw_noise_std" value="0.02" />
    <param name="landmark_match_threshold" value="0.7" />
  </node>

  <!-- PF -->
  <node name="pf_node" pkg="example_package" type="filter_node_PF" output="screen">
    <param name="num_particles" value="100" />
    <param name="world_x_min" value="-10.0" />
    <param name="world_x_max" value="10.0" />
    <param name="world_y_min" value="-10.0" />
    <param name="world_y_max" value="10.0" />
  </node>

  <!-- Logger -->
  <node name="pose_logger" pkg="example_package" type="pose_logger_node" output="screen" />

</launch>


